spec.md

```diff
@@ -0,0 +1,123 @@
+# SBD Spec
+
+This SBD spec defines the protocol for SBD servers and clients to communicate.
+SBD is a simple websocket-based message relay protocol. 
+
+## 1. Websocket Stack
+
+The SBD protocol is built upon websockets.
+
+### 1.1. Websocket Configuration
+
+#### 1.1.1. Message and Frame Size
+
+The maximum SBD message size (including 32 byte header) is 20000 bytes.
+
+SBD clients and servers MAY set the max message size in the websocket library to 20000 to help enforce this.
+
+The maximum frame size MUST be set larger than 20000 so that sbd messages always fit in a single websocket frame.
+
+## 2. Cryptography
+
+### 2.1. TLS
+
+Generally over the WAN SBD servers SHOULD be available over TLS (wss://), and on a LAN without (ws://).
+
+### 2.2. Ed25519
+
+Clients will be identified by ed25519 public key. Client sessions will be validated by ed25519 signature.
+
+## 3. Protocol
+
+### 3.1. Connect Path
+
+Clients MUST specify exactly 1 http path item on the websocket connection url.
+This item must be the base64url encoded public key that this client will be identified by.
+This public key MUST be unique to this new connection.
```

Comment:
Shouldn't this be *SHOULD* rather than a *MUST*? The server can reject incoming public keys if there is already an open connection for it but if the client knows it doesn't have an open connection then there's no harm in opening a new connection with a public key they've used before

Reply:
:+1: